lib/elixir/src/elixir_clauses.erl

```diff
@@ -298,6 +303,14 @@ assert_at_most_once(Kind, [{Kind, _} | Rest], Count, Fun) ->
 assert_at_most_once(Kind, [_ | Rest], Count, Fun) ->
   assert_at_most_once(Kind, Rest, Count, Fun).
 
+check_catch_after_rescue([], _, _) -> ok;
```

Comment:
Maybe name this function `warn_catch_before_rescue` instead?
We use such naming in other places, for example `warn_underscored_var_access`, `warn_underscored_var_repeat`, etc.